ricecooker/utils/youtube_cache.py

```diff
@@ -0,0 +1,167 @@
+import json
+import os
+import re
+import youtube_dl
+
+from pressurecooker.youtube import YouTubeResource
+from ricecooker.config import LOGGER
+
+
+# CONSTANTS for YouTube cache
+################################################################################
+CHEFDATA_DIR = 'chefdata'
+DEFAULT_YOUTUBE_CACHE_DIR = os.path.join(CHEFDATA_DIR, 'youtubecache')
+
+
+# CONSTANTS for YouTube resources
+################################################################################
+YOUTUBE_VIDEO_REGEX = re.compile(
+    r'(https?://)?(www\.)?(youtube|youtu|youtube-nocookie)\.(com|be)/(watch\?v=|embed/|v/|.+\?v=)?(?P<youtube_id>[A-Za-z0-9\-=_]{11})'
+)
+YOUTUBE_PLAYLIST_URL_FORMAT = "https://www.youtube.com/playlist?list={0}"
+YOUTUBE_VIDEO_URL_FORMAT = "https://www.youtube.com/watch?v={0}"
+
+
+class YouTubeVideoCache(object):
+
+    def __init__(self, video_id, alias='', cache_dir=''):
```

Comment:
What is the use-case for specifying an alias for the cache name?

Reply:
There might be some potential usage that users might wanna use self-defined cache filename instead of the YouTube ID by default which actually is not straightforward enough. For example, in the Refugee Response project, I use the language code as the cache filename.